alacritty_terminal/src/tty/unix.rs

```diff
@@ -174,7 +178,11 @@ pub fn new<C>(config: &Config<C>, size: &SizeInfo, window_id: Option<usize>) ->
     // Setup shell environment.
     builder.env("LOGNAME", pw.name);
     builder.env("USER", pw.name);
-    builder.env("SHELL", pw.shell);
+    if cfg!(target_os = "macos") {
+        builder.env("SHELL", pw.shell);
+    } else {
+        builder.env("SHELL", &*shell.program);
+    }
```

Comment:
I think putting this block at the end of the shell env setup with a newline separating it and a comment on why this is done might be a little cleaner.